operator/src/main/java/org/bf2/operator/operands/KafkaCluster.java

```diff
@@ -525,10 +525,17 @@ private JvmOptions buildZooKeeperJvmOptions(ManagedKafka managedKafka) {
 
     private KafkaExporterSpec buildKafkaExporter(ManagedKafka managedKafka) {
         ConfigMap configMap = cachedConfigMap(managedKafka, kafkaExporterLoggingConfigMapName(managedKafka));
+        String strimzi = managedKafka.getSpec().getVersions().getStrimzi();
         KafkaInstanceConfiguration config = this.configs.getConfig(managedKafka);
         KafkaExporterSpecBuilder specBuilder = new KafkaExporterSpecBuilder()
                 .withTopicRegex(".*")
                 .withGroupRegex(".*")
+                .withImage(this.overrideManager.getKafkaExporterImage(strimzi).orElse(null))
+                .withNewTemplate()
+                    .withNewPod()
+                        .withImagePullSecrets(imagePullSecretManager.getOperatorImagePullSecrets(managedKafka))
+                    .endPod()
+                .endTemplate()
                 .withResources(config.getExporter().buildResources());
```

Comment:
This will make the creation of the template and pod depend on the presence of pull secrets.

```suggestion
                .withImage(this.overrideManager.getKafkaExporterImage(strimzi).orElse(null))
                .withResources(config.getExporter().buildResources());

        var pullSecrets = imagePullSecretManager.getOperatorImagePullSecrets(managedKafka);
        if (!pullSecrets.isEmpty()) {
            specBuilder.editOrNewTemplate()
                    .editOrNewPod()
                        .withImagePullSecrets(pullSecrets)
                    .endPod()
                .endTemplate();
        }
```